com/cond_test.go

```diff
@@ -0,0 +1,80 @@
+package com
+
+import (
+	"context"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func TestCond_Broadcast(t *testing.T) {
+	cond := NewCond(context.Background())
+	defer func() { _ = cond.Close() }()
+
+	done := cond.Done()
+	wait := cond.Wait()
+
+	select {
+	case <-done:
+		require.Fail(t, "cond should not be closed, yet")
+	case <-wait:
+		require.Fail(t, "cond should not be ready, yet")
+	case <-time.After(time.Second / 10):
+	}
+
+	cond.Broadcast()
+
+	select {
+	case <-done:
+		require.Fail(t, "cond should still not be closed")
+	case <-cond.Done():
+		require.Fail(t, "cond should not be closed for round 2, yet")
```

Comment:
Is there a difference between those two cases? Both cases check the value of `cond.Done()`, returning `cond.done` which doesn't change over the time. Unless I'm missing something, this is the exact same channel.

Reply:
What you digged out is an implementation detail which tests don't rely on.